ProjFS.Mac/PrjFSKext/PrjFSKext/PerformanceTracing.cpp

```diff
@@ -59,10 +81,23 @@ void PerfTracing_RecordSample(PrjFSPerfCounter counter, uint64_t startTime, uint
             while (interval > oldMax && !atomic_compare_exchange_weak(&result->max, &oldMax, interval))
             {}
         }
+        
+        int intervalLog2 = Log2(interval);
+        atomic_fetch_add(&result->sampleBuckets[intervalLog2], 1);
     }
 }
 
-void InitProbe(PrjFSPerfCounter counter)
+static void InitProbe(PrjFSPerfCounter counter)
 {
     s_perfCounterResults[counter] = PrjFSPerfCounterResult{ .min = UINT64_MAX };
 }
+
+// Computes the floor of the base-2 logarithm of the provided positive integer.
```

Comment:
@jrbriggs I'm hoping this sheds some more light on the calculation being performed here.